scripts/url-registration.ts

```diff
@@ -0,0 +1,182 @@
+  },
+  // PUT /consents/{ID}/error
+  {
+    targetServiceId: 'central-ledger',
+    type: 'THIRDPARTY_CALLBACK_URL_CONSENT_PUT_ERROR',
+    endpointPath: '/consents/{{consentId}}/error',
+  },
+  // POST /authorizations
+  {
+    targetServiceId: 'central-ledger',
+    type: 'FSPIOP_CALLBACK_URL_TRX_REQ_SERVICE',
+    endpointPath: '/authorizations',
+  },
+  // PUT /transfers/{ID}
+  {
+    targetServiceId: 'central-ledger',
+    type: 'FSPIOP_CALLBACK_URL_TRANSFER_PUT',
+    endpointPath: '/transfers/{{transferId}}',
+  },
+  // PUT /transfers/{ID}/error
+  {
+    targetServiceId: 'central-ledger',
+    type: 'FSPIOP_CALLBACK_URL_TRANSFER_PUT_ERROR',
+    endpointPath: '/transfers/{{transferId}}/error',
+  }
+]
+
+program.version('0.0.1')
+
+program.requiredOption(
+  '-h, --host',
+  'Host name for the pisp, which includes the transport protocol. \
+  The host name will be appended with the endpoint paths that are \
+  specified on the Mojaloop OpenAPI specification. For example,   \
+  if the host is "https://example.com" and there is an endpoint   \
+  with path "/participants", then it means the PISP needs to serve\
+  the following endpoint: "https://example.com/participants".',
+  // example value to avoid being considered as boolean
+  'https://pisp-demo-server.local',
+)
+
+program.requiredOption(
+  '-p, --participant-id',
+  'Participant ID of the PISP that is registered in Mojaloop.',
+  // example value to avoid being considered as boolean
+  'pisp',
+)
+
+const requiredServiceIds = Array.from(
+  // Get unique list of services that are needed to the the url
+  // registation.
+  new Set(registrationValues.map((value) => value.targetServiceId))
+)
+
+requiredServiceIds.forEach((serviceId) => {
+  program.requiredOption(
+    '--ml-' + serviceId,
+    'Host name of the ' + serviceId + ' service in Mojaloop. \
+    The value should also include the transport protocol.    \
+    PISP endpoints that are related to the service will be   \
+    registered by sending the registration object to the     \
+    given address. Example: "https://central-ledger.local".',
+    // example value to avoid being considered as boolean
+    'https://' + serviceId + '.local',
+  )
+})
+
+program.parse(process.argv)
+
+// Replaces dash separated string to camel case.
+// This function is needed as the commander library changes
+// the command line argument into a camel case variable. 
+// For example, to get the value of '--ml-central-ledger'
+// argument, we need to access it as `program.mlCentralLedger`.
+function toCamelCase(str: string) {
+  return str.toLowerCase().replace(/-(.)/g, (_, group) => {
+    return group.toUpperCase();
+  })
+}
+
+registrationValues.forEach(async (value) => {
+  // Compute variable name after being altered by the commander library
+  const variableName = toCamelCase('ml-' + value.targetServiceId)
+
+  // Target URL in Mojaloop
+  const targetUrl = program[variableName] + '/participants/'
```

Comment:
It's nicer to use `${program[variableName]}/participants/${program.participantId}/endpoints` string formatting